### PR TITLE
Extract findRoute to room

### PR DESCRIPTION
**Pull request by TooAngel**
For nextroomers and helpers there could be no route to the target.
So check if a route is available.

### CODE REVIEW

**Comment by democratic-collaboration** [comment]
[democratic collaboration](https://github.com/TooAngel/democratic-collaboration)
Approved reviews will speed up the merge, request changes will slow it down.

Please review the PR to help.

Summoning some reviewers:
 - @Nevrdid: A proper review is appreciated
 - @Somotaw: Please look at the code
 - @WhiteHalmos: Not sure why you are here

File: src/prototype_room_my.js [hunk 1]
```diff
@@ -540,29 +540,24 @@ Room.prototype.reviveRoom = function() {
...
-              targetRoom: this.name
-            }
-          });
-        }
-        roomOther.log('Queuing ' + creepToSpawn.role + ' for ' + this.name);
-        if (!this.inQueue(creepToSpawn)) {
-          roomOther.memory.queue.push(creepToSpawn);
+          roomOther.checkRoleToSpawn('defender', 1, undefined, this.name);
         }
+        roomOther.log('Queuing ' + role + ' for ' + this.name);
```
Comment (Nevrdid) [inline src/prototype_room_my.js]: not sure you want this , role is only for nextroomers. plus there is already a log into checkRoleToSpawn, or deeper. i placed one for queing and one for spawning

File: src/prototype_room_my.js [hunk 2]
```diff
@@ -540,29 +540,24 @@ Room.prototype.reviveRoom = function() {
...
-          role: 'nextroomer',
-          routing: {
-            targetRoom: this.name
-          }
-        };
+        let route = this.findRoute(roomOther.name, this.name);
+        // TODO Instead of skipping we could try to free up the way: nextroomerattack or squad
+        if (route.length === 0) {
+          roomOther.log('No route to other room: ' + roomOther.name);
+          continue;
```
Comment (Nevrdid) [inline src/prototype_room_my.js]: will it not skip if route blocked ?

File: src/prototype_room_my.js [hunk 3]
```diff
@@ -540,29 +540,24 @@ Room.prototype.reviveRoom = function() {
...
+        }
+
+        let role = 'nextroomer';
         if (this.memory.wayBlocked) {
-          creepToSpawn.role = 'nextroomerattack';
+          role = 'nextroomerattack';
```
Comment (Nevrdid) [inline src/prototype_room_my.js]: role = this.memory.wayBlocked ? 'nextroomerattack' : 'nextroomer'

File: src/prototype_room_my.js (hunk 1 above)
Comment (Nevrdid) [inline src/prototype_room_my.js]: there is already a log into checkRoleToSpawn, or deeper. i placed one for queing and one for spawning

File: src/prototype_room_my.js (hunk 2 above)
Comment (TooAngel) [inline src/prototype_room_my.js]: I don't understand the question.

I had cases where a route couldn't be calculated and a creep was spawned.

File: src/prototype_room_my.js (hunk 3 above)
Comment (TooAngel) [inline src/prototype_room_my.js]: k, thx

File: src/prototype_room_my.js (hunk 1 above)
Comment (TooAngel) [inline src/prototype_room_my.js]: k

File: src/prototype_room_my.js (hunk 2 above)
Comment (Nevrdid) [inline src/prototype_room_my.js]: ok, got it. it should be used so i supose in canHelp too i supose as i did a check for dist < cst. you can change dist value using getRoute

Comment (Nevrdid) [inline src/prototype_room_my.js]: https://github.com/TooAngel/screeps/blob/master/src/prototype_room_my.js#L226
https://github.com/TooAngel/screeps/blob/findroute/src/prototype_room_utils.js#L13

**Pull request merged**